src/CachePeer.cc

```diff
@@ -114,6 +117,206 @@ CachePeer::connectTimeout() const
+
+    if (options.default_parent)
+        yaml << " default";
+
+    if (options.roundrobin)
+        yaml << " round-robin";
+
+    if (options.carp)
+        yaml << " carp";
+
+#if USE_AUTH
+    if (options.userhash)
+        yaml << " userhash";
+#endif
+
+    if (options.sourcehash)
+        yaml << " sourcehash";
+
+    if (options.weighted_roundrobin)
+        yaml << " weighted-round-robin";
+
+    if (options.mcast_responder)
+        yaml << " multicast-responder";
+
+#if PEER_MULTICAST_SIBLINGS
+    if (options.mcast_siblings)
+        yaml << " multicast-siblings";
+#endif
+
+    if (weight != 1)
+        yaml << " weight=" << weight;
+
+    if (options.closest_only)
+        yaml << " closest-only";
+
+#if USE_HTCP
+    if (options.htcp)
+    {
+        std::vector<const char *> opts {};
+        if (options.htcp_oldsquid)
+            opts.push_back("oldsquid");
+        if (options.htcp_no_clr)
+            opts.push_back("no-clr");
+        if (options.htcp_no_purge_clr)
+            opts.push_back("no-purge-clr");
+        if (options.htcp_only_clr)
+            opts.push_back("only-clr");
+        yaml << AsList(opts).prefixedBy(" htcp=").delimitedBy(",");
+    }
+#endif
+
+    if (options.no_netdb_exchange)
+        yaml << " no-netdb-exchange";
+
+#if USE_DELAY_POOLS
+    if (options.no_delay)
+        yaml << " no-delay";
+#endif
+
+    if (login)
+        yaml << " login=" << login;
+
+    if (mcast.ttl > 0)
+        yaml << " ttl=" << mcast.ttl;
+
+    if (connect_timeout_raw > 0)
+        yaml << " connect-timeout=" << connect_timeout_raw;
+
+    if (connect_fail_limit != PEER_TCP_MAGIC_COUNT)
+        yaml << " connect-fail-limit=" << connect_fail_limit;
+
+#if USE_CACHE_DIGESTS
+    if (digest_url)
+        yaml << " digest-url=" << digest_url;
+#endif
+
+    if (options.allow_miss)
+        yaml << " allow-miss";
+
+    if (options.no_tproxy)
+        yaml << " no-tproxy";
+
+    if (max_conn > 0)
+        yaml << " max-conn=" << max_conn;
+
+    if (standby.limit > 0)
+        yaml << " standby=" << standby.limit;
+
+    if (options.originserver)
+        yaml << " originserver";
+
+    if (domain)
+        yaml << " forceddomain=" << domain;
+
+    if (connection_auth == 0)
+        yaml << " connection-auth=off";
+    else if (connection_auth == 1)
+        yaml << " connection-auth=on";
+    else if (connection_auth == 2)
+        yaml << " connection-auth=auto";
+
+    secure.dumpCfg(yaml, "tls-");
+}
+
+void
+CachePeer::reportStatistics (std::ostream& yaml)
+{
+    yaml <<
+       name << ":\n" <<
+       "  type: " << typeString() << '\n' <<
+       "  host: " << host << '/' << http_port << '/' << icp.port << '\n';
+    yaml << "  flags:";
+    dumpOptions(yaml);
+    yaml << '\n';
+
+    std::vector<Ip::Address> addr;
+    std::copy(addresses, addresses+n_addresses, std::back_inserter(addr));
+    yaml << "  addresses: [ " << AsList(addr).prefixedBy("\"").delimitedBy("\", \"").suffixedBy("\"") << " ]\n";
```

Comment:
The prefix and suffix can just be hard-coded with the brackets to make this simpler.
```suggestion
    yaml << "  addresses: [ \"" << AsList(addr).delimitedBy("\", \"") << "\" ]\n";
```

Reply:
Not really, that would turn the case where the list of addresses is empty to an empty string

Reply:
This unreadable code code needs to be improved, but the correct fix is different. I did not say anything in my review because there are several overlapping problems in these two lines that may warrant a prerequisite PR if we start addressing them.

If you want to start moving in that direction, add AsList::quoted(void) or AsList::quotedValues(void).

```suggestion
    yaml << "  addresses: [ " << AsList(addr).quoted().delimitedBy(", ") << " ]\n";
```

Reply:
PR #1779 implements AsList::quotedBy()

Waiting for that to be reviewed and merged

Reply:
Started using quoted() early in commit 1d10221d03, resolving